src/scripts/booking.js

```diff
@@ -0,0 +1,191 @@
+const api = 'https://lernia-sjj-assignments.vercel.app/api';
+const body = document.body;
+
+let challengeTitle;
+let challengeNumber;
+let times;
+let date;
+let finalRes;
+
+// Close Function, the booking process stops if you click outside the form
+function addCancelOption() {
+  const close = document.querySelector(".booking");
+  close.addEventListener('click', (e) => {
+    if(e.target.className == "booking"){
+      challengeTitle = "";
+      challengeNumber = "";
+      times = "";
+      date = "";
+      body.removeChild(close);
+      body.style.overflow = "auto";
+      return;
+    }
+  });
+}
+
+// Book a room (First Step)
+function book(title, challenge) {
+  challengeTitle = title;
+  challengeNumber = challenge;
+
+  const firstTemplate = `
+  <div class="booking" id="bookingPartOne"> 
+    <div class="first-modal">
+      <h2>Book room "${title}" (Step 1)</h2>
+      <p>What date would you like to come?</p>
+      <label for="date">Date</label>
+      <input name="date" type="date" id="bookingDate">
+      <div class="btn-parent">
+          <button id="submitBtnOne" class="button primary">Search available times</button>
+      </div>
+    </div>
+  </div>
+  `;
+  
+  body.insertAdjacentHTML("afterbegin", firstTemplate);
+  body.style.overflow = "hidden";
+
+  // Submit
+  const submitBtn = document.querySelector("#submitBtnOne");
+  submitBtn.addEventListener('click', () => {
+    submitStepOne(challenge);
+  });
+
+  addCancelOption();
+}
+
+// Submit Step 1
+async function submitStepOne(challenge) {
+  date = document.querySelector("#bookingDate").value;
+
+  if (!date) {
+    alert('Please fill the date field!');
+    return;
+  }
+
+  const res = await fetch(`${api}/booking/available-times?date=${date}&challenge=${challenge}`);
+  times = await res.json();
+  times = times.slots;
+
+  if(!times) {
+    alert('That date is not available, try another one!')
+    return;
+  }
+
+  bookStepTwo();
+}
+
+// Book a room (Second Step)
+function bookStepTwo() {
+  const bookingPartOneDiv = document.querySelector('#bookingPartOne');
+  body.removeChild(bookingPartOneDiv);
+
+  const secondTemplate = `
+  <div class="booking" id="bookingPartTwo">
+    <div class="second-modal">
+        <h2>Book room "${challengeTitle}" (Step 2)</h2>
+
+        <label for="name">Name</label>
+        <input type="text" name="name" id="name">
+
+        <label for="email">E-mail</label>
+        <input type="email" name="email" id="email">
+
+        <label for="time">What time?</label>
+        <select name="time" id="time"></select>
+
+        <label for="participants">How many participants?</label>
+        <select name="participants" id="participants">
```

Comment:
As mentioned earlier, the max and min number of participants should be loaded from the data so when you book you get the right number of participants.

Reply:
👍